### PR TITLE
feat(consistent-cards-ui): Add Start over button to automated checks

### PR DESCRIPTION
**Pull request by DaveTryon**
#### Description of changes

Implement the __Start over__ button for FastPass with cards UI (works for both automated checks and tabstops). Assessment view still uses the StartOverDropdown; FastPass view uses an ActionButton. When clicked, the ActionButton uses a method on the 

Earlier code in this feature added the `StartOverComponentPropertyFactory` and `ReportExportComponentPropertyFactory` classes, which returned property objects, then relied upon other code to build the object. This PR changes that approach so the factories (now named `StartOverComponentFactory` and `ReportExportComponentFactory`) each return a JSX.Element object that can be consumed directly. This brings the property and creation code into a single location for each type. It also reduces overlap between tests

#### Pull request checklist

- [x] Addresses an existing issue: 1638935
- [x] Ran `yarn fastpass`
- [x] Added/updated relevant unit test(s) (and ran `yarn test`)
- [x] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [x] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). Check workflow guide at: `<rootDir>/docs/workflow.md`
- [pending] (UI changes only) Added screenshots/GIFs to description above
- [pending] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/DetailsView/components/report-export-component-factory.tsx [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    const reportGenerator = deps.reportGenerator;
+
+    const reportExportComponentProps: ReportExportComponentProps = {
+        deps: deps,
+        scanDate: scanDate,
+        reportGenerator: reportGenerator,
+        pageTitle: tabStoreData.title,
+        exportResultsType: 'AutomatedChecks',
+        htmlGenerator: description =>
+            reportGenerator.generateFastPassAutomateChecksReport(
```
Comment (DaveTryon) [inline src/DetailsView/components/report-export-component-factory.tsx]: The method being called here should arguably be renamed to `generateFastPassAutomatedChecksReport`, but that will be a separate PR

File: src/DetailsView/components/start-over-component-factory.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        return null;
+    }
+
+    const selectedTest = props.visualizationStoreData.selectedFastPassDetailsView;
+    const actionMessageCreator = props.actionMessageCreator;
+
+    return (
+        <ActionButton
+            iconProps={{ iconName: 'Refresh' }}
+            onClick={(event: SupportedMouseEvent) => onClickFastPass(event, actionMessageCreator, selectedTest)}
```
Comment (waabid) [inline src/DetailsView/components/start-over-component-factory.tsx]: You can actually do something like this here:

```(event: SupportedMouseEvent) => actionMessageCreator.rescanVisualization(event, selectedTest)```

File: src/tests/unit/tests/DetailsView/components/start-over-component-factory.test.tsx [hunk 3]
```diff
@@ -100,12 +103,22 @@ describe('StartOverComponentPropsFactory', () => {
...
         const props = getProps(false);
         const rendered = getStartOverComponentForFastPass(props);
 
         expect(rendered).toMatchSnapshot();
     });
+
+    test('getStartOverComponentPropsForFastPass, CardsUI is true, scanResults is not null, scanning is true, component matches snapshot', () => {
+        setScanResult();
+        setCardsUiFlag(true);
+        scanning = 'true';
```
Comment (waabid) [inline src/tests/unit/tests/DetailsView/components/start-over-component-factory.test.tsx]: Just a heads up we don't usually set scanning to "true"; the string is usually a string identifier for a test.

**Comment by waabid** [comment]
Just a heads up we don't usually set scanning to "true"; the string is usually a string identifier for a test.

So it alternates between being null (nothing scanning) and scanning (where it carries the value of what's being scanned).
